pkgs/development/compilers/flutter/default.nix

```diff
@@ -17,12 +17,34 @@ let
           url = "https://storage.googleapis.com/dart-archive/channels/stable/release/${dartVersion}/sdk/dartsdk-linux-arm64-release.zip";
           sha256 = dartHash.aarch64-linux;
         };
+        "${dartVersion}-x86_64-darwin" = fetchurl {
+          url = "https://storage.googleapis.com/dart-archive/channels/stable/release/${dartVersion}/sdk/dartsdk-macos-x64-release.zip";
+          sha256 = dartHash.x86_64-darwin;
+        };
+        "${dartVersion}-aarch64-darwin" = fetchurl {
+          url = "https://storage.googleapis.com/dart-archive/channels/stable/release/${dartVersion}/sdk/dartsdk-macos-arm64-release.zip";
+          sha256 = dartHash.aarch64-darwin;
+        };
       };
     };
-    src = fetchurl {
-      url = "https://storage.googleapis.com/flutter_infra_release/releases/stable/linux/flutter_linux_${version}-stable.tar.xz";
-      sha256 = hash;
-    };
+    src = rec {
+      x86_64-linux = fetchurl {
+        url = "https://storage.googleapis.com/flutter_infra_release/releases/stable/linux/flutter_linux_${version}-stable.tar.xz";
+        sha256 = flutterHash.x86_64-linux;
+      };
+      aarch64-linux = fetchurl {
+        url = "https://storage.googleapis.com/flutter_infra_release/releases/stable/linux/flutter_linux_${version}-stable.tar.xz";
+        sha256 = flutterHash.aarch64-linux;
+      };
+      x86_64-darwin = fetchzip {
```

Comment:
Can `fetchurl` be used here?

Reply:
I tried to replace `fetchzip` with `fetchurl`, but the following error came out 🤔 

```
unpacking sources
unpacking source archive /nix/store/njydk0l4j8xjlah76a1261dbsrmn8drz-flutter_macos_3.3.8-stable.zip
do not know how to unpack source archive /nix/store/njydk0l4j8xjlah76a1261dbsrmn8drz-flutter_macos_3.3.8-stable.zip
error: builder for '/nix/store/qscx3g5riw142lsj75adiv3pimliprp1-flutter-3.3.8-unwrapped.drv' failed with exit code 1;
       last 3 log lines:
       > unpacking sources
       > unpacking source archive /nix/store/njydk0l4j8xjlah76a1261dbsrmn8drz-flutter_macos_3.3.8-stable.zip
       > do not know how to unpack source archive /nix/store/njydk0l4j8xjlah76a1261dbsrmn8drz-flutter_macos_3.3.8-stable.zip
```

Reply:
Oh, I guess you need to add `unzip` to `nativeBuildInputs`.

Reply:
Seems working well 👍 Thanks @wegank !